Index: nss-3.20/nss/lib/freebl/drbg.c
===================================================================
--- a/nss/lib/freebl/drbg.c
+++ b/nss/lib/freebl/drbg.c
@@ -508,7 +508,11 @@ RNG_RandomUpdate(const void *data, size_
      *   NS_PTR_GT_32 will correct that mistake.
      */
 
+#if defined(NSS_USE_64)
+    PR_STATIC_ASSERT(sizeof(size_t) <= 4);
+#else
     PR_STATIC_ASSERT(sizeof(size_t) > 4);
+#endif
 
     if (bytes > PRNG_MAX_ADDITIONAL_BYTES) {
 	bytes = PRNG_MAX_ADDITIONAL_BYTES;
