Index: libffi-3.2.1/configure
===================================================================
--- a/configure
+++ b/configure
@@ -18725,11 +18725,6 @@ if test "x$GCC" = "xyes"; then
     toolexecdir="${libdir}"/gcc-lib/'$(target_alias)'
     toolexeclibdir="${libdir}"
   fi
-  multi_os_directory=`$CC $CFLAGS -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
 
 else
   toolexeclibdir="${libdir}"
Index: libffi-3.2.1/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -601,11 +601,6 @@ if test "x$GCC" = "xyes"; then
     toolexecdir="${libdir}"/gcc-lib/'$(target_alias)'
     toolexeclibdir="${libdir}"
   fi
-  multi_os_directory=`$CC $CFLAGS -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
   AC_SUBST(toolexecdir)
 else
   toolexeclibdir="${libdir}"
