Index: alljoyn-15.09.00-src/alljoyn_core/router/posix/SConscript
===================================================================
--- alljoyn-15.09.00-src.orig/alljoyn_core/router/posix/SConscript
+++ alljoyn-15.09.00-src/alljoyn_core/router/posix/SConscript
@@ -19,6 +19,7 @@ Import('router_env', 'router_objs', 'sro
 if router_env['OS'] != 'darwin':
     daemon_env = router_env.Clone()
     if daemon_env['OS'] != 'android':
+	daemon_env.Append(LIBS = 'attr'),
         daemon_env.Append(LIBS = 'cap')
     daemon = daemon_env.Program('alljoyn-daemon', ['daemon-main.cc'] + srobj + router_objs)
 else:
